modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -132,13 +132,21 @@ public function add_browse_sites_link() {
 	 * Adds a link to the menu to create a new site.
 	 */
 	public function add_new_site_link() {
-		// Only show the menu on atomic or simple.
-		if ( ! jetpack_is_atomic_site() && ! $this->is_wpcom_site() ) {
+		if ( jetpack_is_atomic_site() ) {
+			$wpcom_user_data = ( new Connection_Manager() )->get_connected_user_data();
+
+			if ( $wpcom_user_data['site_count'] > 1 ) {
+				return;
+			}
+		} elseif ( is_multisite() && count( get_blogs_of_user( get_current_user_id() ) ) > 1 ) {
 			return;
 		}
 
-		// Add the menu item.
-		add_menu_page( __( 'Add new site', 'jetpack' ), __( 'Add new site', 'jetpack' ), 'read', 'https://wordpress.com/start', null, 'dashicons-plus-alt', 98 );
+		// Get last position.
+		$position = key( array_slice( $GLOBALS['menu'], -1 ) );
```

Comment:
This is actually adding the menu after `My Home`
![](https://cln.sh/qvJBMG+) 
cause at that time this is the last position 
https://github.com/Automattic/jetpack/blob/3285b0aa4bbf15a7121ac2a3476bddc2d81e8098/modules%2Fmasterbar%2Fadmin-menu%2Fclass-admin-menu.php#L73-L77

Can't we do `$position = 1000` or something similar?

Reply:
@frontdevde could you weigh in here? If we resolve this, I think we can merge it and iterate in the future regarding the landing URL and the styles.

Reply:
This seems like a good solution

$position = 1000;
while ( isset( $menu[ $position ] ) ) {
  $position++;
}

Reply:
Not a huge fan of adding a somewhat arbitrary high value. I'd prefer we had a more reliable solution to tell what the highest position value is when rendering the menu. That said the solution you propose solves it for now.

Reply:
Pushed the commits to fix that.